link.go

```diff
@@ -334,9 +367,10 @@ Loop:
 					// decrement link-credit after entire message transferred
 					if !tr.More {
 						l.deliveryCount++
-						l.linkCredit--
+
+						newCredit := atomic.AddUint32(&l.linkCredit, ^uint32(0))
```

Comment:
Why did this need to change to atomic decrement?  And to follow up, do we need to perform atomic increment and/or atomic in other places too?

Reply:
I actually can probably revert this.

I had two different models I'd gone through with this - one was where I modified `.linkCredit` directly and this final one where I just have an accumulated value. Since I no longer touch linkCredit (it's still only done in the event loop) I should just be able to get rid of these.

Reply:
(and there are some other spots as well - I'll basically just revert the lines that modify `.linkCredit`)

Reply:
Reverted!